squirrel.worker/src/main/java/org/dice_research/squirrel/worker/impl/WorkerImpl.java

```diff
@@ -328,15 +330,17 @@ public void sendNewUris(Iterator<byte[]> uriIterator) {
                 uriProcessor.recognizeUriType(newUri);
                 newUris.add(newUri);
                 if ((newUris.size() >= (packageCount + 1) * MAX_URIS_PER_MESSAGE) && uriIterator.hasNext()) {
+                    List<CrawleableUri> distinctUris = newUris.stream().distinct().collect(Collectors.toList());
                     frontier.addNewUris(
-                            new ArrayList<>(newUris.subList(packageCount * MAX_URIS_PER_MESSAGE, newUris.size())));
+                            new ArrayList<>(distinctUris.subList(packageCount * MAX_URIS_PER_MESSAGE, newUris.size())));
                     packageCount++;
                 }
             } catch (Exception e) {
                 LOGGER.warn("Couldn't handle the (de-)serialization of a URI. It will be ignored.", e);
             }
         }
-        frontier.addNewUris(newUris);
+        List<CrawleableUri> distinctUris = newUris.stream().distinct().collect(Collectors.toList());
+        frontier.addNewUris(distinctUris);
```

Comment:
The same as above: calling the distinct method does not help and will consume a lot of runtime.